audiostream2py/reader.py

```diff
@@ -0,0 +1,61 @@
+"""BufferReader specific to AudioData"""
+
+from typing import List
+
+from stream2py import BufferReader
+
+from audiostream2py.data import AudioData
+
+
+class OverlapsPastError(IndexError):
+    pass
+
+
+class OverlapsFutureError(IndexError):
+    pass
+
+
+class AudioBufferReader(BufferReader):
```

Comment:
We can reuse `AudioDataGetter` here.

Reply:
That's not possible. The internal buffer and how it's queried is different.  It would be easier to reuse `AudioBufferReader` in `AudioDataGetter`.  That would involve removing `creek`.